### PR TITLE
feat(Authoring): Add delete button to all steps

### LINKED ISSUE #1631: feat(Authoring): In project view, add delete button to steps
Also add a border around each step.

### PR DESCRIPTION
**Pull request by geoffreykwan**
## Changes
- Added delete button to the steps
- The button only show up when mousing over the step
- Added a border around each step

## Test
Make sure the delete button deletes the step

Closes #1631

### CODE REVIEW

**Review by hirokiterashima — CHANGES_REQUESTED** [review]
This is a lot of changes for me to review at once. Can you break it up into smaller PR's? Maybe we should create a feature branch?

**Comment by geoffreykwan** [comment]
I removed the move and copy buttons.

File: src/assets/wise5/authoringTool/project-authoring-step/project-authoring-step.component.ts [hunk 1]
```diff
@@ -4,13 +4,15 @@ import { TeacherDataService } from '../../services/teacherDataService';
...
 import { NodeTypeSelected } from '../domain/node-type-selected';
+import { DeleteNodeService } from '../../services/deleteNodeService';
 
 @Component({
   selector: 'project-authoring-step',
   templateUrl: './project-authoring-step.component.html',
   styleUrls: ['./project-authoring-step.component.scss']
 })
 export class ProjectAuthoringStepComponent {
+  @Output() deleteEvent: EventEmitter<any> = new EventEmitter<any>();
```
Comment (hirokiterashima) [inline src/assets/wise5/authoringTool/project-authoring-step/project-authoring-step.component.ts]: Is this line necessary? Looks like it's not being used.

File: src/assets/wise5/authoringTool/project-authoring-step/project-authoring-step.component.ts [hunk 2]
```diff
@@ -90,4 +93,17 @@ export class ProjectAuthoringStepComponent {
     this.dataService.setCurrentNodeByNodeId(nodeId);
     this.router.navigate([`/teacher/edit/unit/${this.projectId}/node/${nodeId}/advanced/path`]);
   }
+
+  protected delete(event: any): void {
```
Comment (hirokiterashima) [inline src/assets/wise5/authoringTool/project-authoring-step/project-authoring-step.component.ts]: Add type
```suggestion
  protected delete(event: Event): void {
```

**Review by hirokiterashima — CHANGES_REQUESTED** [review]
Looking good.

Here's what we discussed today:
1. When the "Delete step" tooltip is shown, hide the "Click to enter step" tooltip.
2. Allow authors to delete lessons. The delete icon should appear to the right of the edit icon.

I added inline comments for code improvement suggestions.

**Comment by geoffreykwan** [comment]
I fixed the tooltips and cleaned up the code. We can add the delete lesson button in another issue.

**Review by hirokiterashima — APPROVED** [review]
LGTM.

In the future, it would be good if you could also say why you decided to postpone adding the Delete Button to lesson, because this is different from what we discussed together. Was it because it's too much work, will increase the review scope by a lot, etc.

**Pull request merged**